File: src/test/java/RegisterPage.py
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from BasePage import BasePage


class AuthPage(BasePage):
    FIRSTNAME_FIELD = (By.XPATH, "//input[@name='firstname']")
    LASTNAME_FIELD = (By.XPATH, "//input[@name='lastname']")
    EMAIL_FIELD = (By.XPATH, "//*[@id='input-email']")
    PASSWORD_FIELD = (By.XPATH, "//*[@id='input-password']")
    LOGIN_BUTTON = (By.XPATH, "//*[@id='form-login']/div[3]/button")
    CONTINUE_BUTTON = (By.XPATH, "//button[text()='Continue']")
    AGREE_BUTTON = (By.XPATH, "//input[@name='agree']")

    def __init__(self, driver):
        super().__init__(driver)
        self.driver = driver

    def enter_firstname(self, firstname):
        self.driver.find_element(*self.FIRSTNAME_FIELD).send_keys(firstname)

    def enter_lastname(self, lastname):
        self.driver.find_element(*self.LASTNAME_FIELD).send_keys(lastname)

    def enter_email(self, email):
        self.driver.find_element(*self.EMAIL_FIELD).send_keys(email)

    def enter_password(self, password):
        self.driver.find_element(*self.PASSWORD_FIELD).send_keys(password)

    def click_login(self):
        self.driver.find_element(*self.LOGIN_BUTTON).click()

    def click_agree(self):
        self.driver.find_element(*self.AGREE_BUTTON).send_keys(Keys.SPACE)

    def click_continue(self):
        self.driver.find_element(*self.CONTINUE_BUTTON).click()

File: src/test/java/RegisterPage.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class RegisterPage extends BasePage {
    protected static final By FIRSTNAME_FIELD = By.xpath("//input[@name='firstname']");
    protected static final By LASTNAME_FIELD = By.xpath("//input[@name='lastname']");
    protected static final By EMAIL_FIELD = By.xpath("//*[@id='input-email']");
    protected static final By PASSWORD_FIELD = By.xpath("//*[@id='input-password']");
    protected static final By LOGIN_BUTTON = By.xpath("//*[@id='form-login']/div[3]/button");
    protected static final By CONTINUE_BUTTON = By.xpath("//button[text()='Continue']");
    protected static final By AGREE_BUTTON = By.xpath("//input[@name='agree']");

    public RegisterPage(WebDriver driver) {
        super(driver);
        logger.info("Инициализирована страница");
    }

    public void enterFirstname(String firstname) {
        driver.findElement(FIRSTNAME_FIELD).sendKeys(firstname);

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}({})", methodName, firstname);
    }

    public void enterLastname(String lastname) {
        driver.findElement(LASTNAME_FIELD).sendKeys(lastname);

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}({})", methodName, lastname);
    }

    public void enterEmail(String email) {
        driver.findElement(EMAIL_FIELD).sendKeys(email);

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}({})", methodName, email);
    }

    public void enterPassword(String password) {
        driver.findElement(PASSWORD_FIELD).sendKeys(password);

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}({})", methodName, password);
    }

    public void clickLogin() {
        driver.findElement(LOGIN_BUTTON).click();

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}", methodName);
    }

    public void clickAgree() {
        driver.findElement(AGREE_BUTTON).click();

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}", methodName);
    }

    public void clickContinue() {
        driver.findElement(CONTINUE_BUTTON).click();

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}", methodName);
    }
}